src/routes/home.tsx

```diff
@@ -5,27 +5,47 @@ import Spinner from '@/components/spinner'
 import { compareEvents } from '@/utils/compare-events'
 import { useGeoLocation } from '@/utils/use-geo-location'
 import { useEvents } from '@/cache/events/use'
+import { useState } from 'preact/hooks'
+import { ProcessedEventInfo } from '@/api/event-info'
 
 const now = new Date()
-
 const Home = () => {
   const events = useEvents()
   const location = useGeoLocation()
+  const [term = '', setTerm] = useState('');
+  const lowerCaseQuery = term.toString().toLowerCase();
+
+  function searchingFor(term: string) {
+    return function (event: ProcessedEventInfo) {
+      if (!term) return true;
+      return (event.name.toLowerCase().includes(lowerCaseQuery)) ||
+        (event.key.toLowerCase().includes(lowerCaseQuery)) ||
+        (event.locationName.toLowerCase().includes(lowerCaseQuery)) ||
+        (event.district !== undefined && event.district.toLowerCase().includes(lowerCaseQuery)) ||
+        (event.fullDistrict !== undefined && event.fullDistrict.toLowerCase().includes(lowerCaseQuery));
+    }
+  }
+
   return (
     <Page name="Home" back={false}>
+      <div>Search bar</div>
+      <div class="searchBar">
+        <input type="text" onInput={(e: any) => setTerm(e.target.value)} />
+      </div>
       <div>
         {events ? (
           events
+            .filter(searchingFor(term))
```

Comment:
@marek-jarab Can you inline the `searchingFor` function here?

Reply:
What do you mean by that? because when I run "npm run lint" it organized rows as they used to be.
![rows](https://user-images.githubusercontent.com/55954157/67510421-35855000-f695-11e9-9a54-af575877a5e5.png)

Reply:
I mean that instead of defining the `searchingFor` function outside and calling it here, can you turn it into an inline arrow function here to filter the events?

Reply:
So you want to  remove current function searchingFor() and create new arrow function inside return method with the whole logic? Won't the return() method be  pointlessly too complicated? Also it's seem for me quite complicated to implement , as I am not used to work with arrow functions often

Reply:
Here is what it would look like:

```js
            .filter(event => {
              if (!term) return true
              return (
                event.name.toLowerCase().includes(lowerCaseQuery) ||
                event.key.toLowerCase().includes(lowerCaseQuery) ||
                event.locationName.toLowerCase().includes(lowerCaseQuery) ||
                (event.district !== undefined &&
                  event.district.toLowerCase().includes(lowerCaseQuery)) ||
                (event.fullDistrict !== undefined &&
                  event.fullDistrict.toLowerCase().includes(lowerCaseQuery))
              )
            })
```
I don't think it makes the `return` too complicated

Reply:
Oh Ok. I thought it would be better to create function outside the return method, but this is also good approach.